### PR TITLE
Feat/order page

### PR DESCRIPTION
**Pull request by NataRep**
#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 💡 Background and solution
I'm having a couple of problems with this assignment:
- the output of station names does not work every time
- order status update after deletion happens only after page refresh
if you know how to solve them - write here.

image

#### 🎯 Related issues
[
_Links to related issues_](https://github.com/rolling-scopes-school/tasks/tree/master/tasks/train-a)

### CODE REVIEW

File: src/app/pages/orders-page/orders-page.component.ts [hunk 1]
```diff
@@ -1,8 +1,241 @@
...
+
+    // eslint-disable-next-line @ngrx/avoid-dispatching-multiple-actions-sequentially
+    this.store.dispatch(loadStations());
+
+    this.role$ = this.store.select(selectAccess);
+
+    // eslint-disable-next-line @ngrx/avoid-dispatching-multiple-actions-sequentially
+    this.store.dispatch(loadCarriages());
+    // eslint-disable-next-line @ngrx/avoid-dispatching-multiple-actions-sequentially
+    this.store.dispatch(loadUsers());
```
Comment (mogitrash) [inline src/app/pages/orders-page/orders-page.component.ts]: Remove those (eslint) comments please

File: src/app/pages/orders-page/orders-page.component.ts [hunk 2]
```diff
@@ -1,8 +1,241 @@
...
+        endTime: this.getEndData(order),
+        durationTrip: this.calculateDuration(order),
+        numberCarriage: carriageData.number,
+        typeCarriage: carriageData.type,
+        numberSeat: carriageData.seat,
+        price: this.calculateTotalPrice(order, carriageData.type),
+        status: order.status,
+      };
+    });
+  }
```
Comment (mogitrash) [inline src/app/pages/orders-page/orders-page.component.ts:111]: 👍

**Review by mogitrash — APPROVED** [review]
Good job! 👍

File: src/app/pages/orders-page/orders-page.component.ts (hunk 1 above)
Comment (NataRep) [reply src/app/pages/orders-page/orders-page.component.ts]: I removed the test order comment, but I can't do it without it:

// eslint-disable-next-line @ngrx/avoid-dispatching-multiple-actions-sequentially

File: src/app/pages/orders-page/order/order.component.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  styleUrl: './order.component.scss',
+})
+export class OrderComponent implements OnInit {
+  constructor(private store: Store) {}
+
+  @Input({ required: true }) order!: OrderForView;
+
+  @Output() delete = new EventEmitter<number>();
+
+  startStation!: string;
```
Comment (Arthurchkeee) [inline src/app/pages/orders-page/order/order.component.ts:31]: Please, don't forget about access modifiers

File: src/app/pages/orders-page/orders-page.component.ts (hunk 1 above)
Comment (Aleksa013) [reply src/app/pages/orders-page/orders-page.component.ts]: maybe effect is needed here?

File: src/app/core/guards/admin-user.guard.ts [hunk 4]
```diff
@@ -1,21 +1,14 @@
...
-    take(1),
-    map((currentAccess) => {
-      if (currentAccess === 'manager' || currentAccess === 'user') {
-        return true;
-      }
-      router.navigate(['/signin']);
-      return false;
-    }),
-  );
+  if (token) {
```
Comment (Aleksa013) [inline src/app/core/guards/admin-user.guard.ts:8]: May be used here localStorage and check role?

File: src/app/pages/orders-page/orders-page.component.ts [hunk 5]
```diff
@@ -1,8 +1,232 @@
...
+
+  stations$: Observable<Station[] | null> = this.store.select(selectStations);
+
+  carriages$: Observable<Carriage[] | null> = this.store.select(selectCarriages);
+
+  users$: Observable<User[] | null> = this.store.select(selectUsers);
+
+  role$!: Observable<Access>;
+
+  ordersForView!: OrderForView[];
```
Comment (Arthurchkeee) [inline src/app/pages/orders-page/orders-page.component.ts]: Again 😢

File: src/app/pages/orders-page/orders-page.component.ts [hunk 6]
```diff
@@ -1,8 +1,232 @@
...
+  styleUrls: ['./orders-page.component.scss'],
 })
-export class OrdersPageComponent {}
+export class OrdersPageComponent implements OnInit, OnDestroy {
+  orders$: Observable<Order[] | null> = this.store.select(selectOrders);
+
+  stations$: Observable<Station[] | null> = this.store.select(selectStations);
+
+  carriages$: Observable<Carriage[] | null> = this.store.select(selectCarriages);
+
```
Comment (Aleksa013) [inline src/app/pages/orders-page/orders-page.component.ts:42]: Please, don't forget about access modifiers

File: src/app/pages/orders-page/orders-page.component.ts [hunk 7]
```diff
@@ -1,8 +1,232 @@
...
+    this.role$
+      .pipe(
+        switchMap((role) => {
+          if (role === 'manager') {
+            this.store.dispatch(loadOrders({ all: true }));
+          } else if (role === 'user') {
+            this.store.dispatch(loadOrders({}));
+          }
+          return combineLatest([this.orders$, this.stations$, this.carriages$, this.users$]);
```
Comment (Aleksa013) [inline src/app/pages/orders-page/orders-page.component.ts:67]: great practice rxjs

**Review by Arthurchkeee — CHANGES_REQUESTED** [review]
Please, don't forget about access modifiers

File: src/app/shared/pipes/order-view.pipe.ts [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+import { Order } from '../../features/trips/models/order.model';
+import { OrderForView } from '../../pages/orders-page/order/order.component';
+
+@Pipe({
+  name: 'transformOrder',
+})
+export class TransformOrderPipe implements PipeTransform {
+  // хелперы для преобразования seatId в typeCarriage и numberCarriage
+  private getTypeCarriage(seatId: number): string {
+    // Пример вычислений, заменить на реальные
```
Comment (Aleksa013) [inline src/app/shared/pipes/order-view.pipe.ts]: I think it's better to use English for comments

File: src/app/shared/pipes/order-view.pipe.ts [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    return seatId < 20 ? 'carriage1' : 'carriage2';
+  }
+
+  private getNumberCarriage(seatId: number): number {
+    // Пример вычислений, заменить на реальные
+    // тут нужно получить о количестве мест в вагонах поезда с сервера используя ngrx
```
Comment (Aleksa013) [inline src/app/shared/pipes/order-view.pipe.ts]: Thanks for your comments. I forgot it all the time

**Review by Aleksa013 — APPROVED** [review]
Great job!

File: src/app/pages/orders-page/orders-page.component.ts (hunk 1 above)
Comment (NataRep) [reply src/app/pages/orders-page/orders-page.component.ts]: You're right, I added a new action and a redusser effect for it.

File: src/app/pages/orders-page/order/order.component.ts (hunk 3 above)
Comment (NataRep) [reply src/app/pages/orders-page/order/order.component.ts:31]: I'm sorry I forgot about them again.

Comment (NataRep) [reply src/app/pages/orders-page/order/order.component.ts:31]: Done

File: src/app/pages/orders-page/orders-page.component.ts (hunk 5 above)
Comment (NataRep) [reply src/app/pages/orders-page/orders-page.component.ts]: I corrected it )

File: src/app/core/guards/admin-user.guard.ts (hunk 4 above)
Comment (NataRep) [reply src/app/core/guards/admin-user.guard.ts:8]: Yeah, but that's not the mission of this PR

File: src/app/pages/orders-page/orders-page.component.ts (hunk 6 above)
Comment (NataRep) [reply src/app/pages/orders-page/orders-page.component.ts:42]: I corrected it )

File: src/app/shared/pipes/order-view.pipe.ts (hunk 8 above)
Comment (NataRep) [reply src/app/shared/pipes/order-view.pipe.ts]: I forgot to delete this pip, it's not in use.

File: src/app/shared/pipes/order-view.pipe.ts (hunk 9 above)
Comment (NataRep) [reply src/app/shared/pipes/order-view.pipe.ts]: I forgot to delete this pip, it's not in use.

**Review by Arthurchkeee — APPROVED** [review]
Great job!

**Pull request merged**